network/runtime.go

```diff
@@ -0,0 +1,35 @@
+package network
+
+import "io"
+
+// These interfaces are intended to match those from whyrusleeping/cbor-gen, such that code generated from that
+// system is automatically usable here (but not mandatory).
+type CBORMarshaler interface {
```

Comment:
I'd actually just put these into a separate `cbor` package.

Reply:
SG. `cbor.Marshaler`, `cbor.Er` etc :)

Reply:
or `ser`, `serde`?

Reply:
Going with (Un)Marshaler and Er